tests/test_mocked_responses/common.py

```diff
@@ -223,10 +223,10 @@ def check_reference(
         """Extend base implementation to support cas_number, ec_number, and chemical_name for substances."""
 
         return (
-            self.obj.cas_number == cas_number
-            and self.obj.ec_number == ec_number
-            and self.obj.chemical_name == chemical_name
-            and super().check_reference(record_history_identity, record_guid)
+            (self.obj.cas_number == cas_number and cas_number is not None)
```

Comment:
We don't need all of them to match, we just need at least one to match that isn't None. I think the previous configuration was resulting in false positives if nothing was provided.